smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/UserTuneManager.java

```diff
@@ -0,0 +1,130 @@
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.pep.PepListener;
+import org.jivesoftware.smackx.pep.PepManager;
+import org.jivesoftware.smackx.pubsub.EventElement;
+import org.jivesoftware.smackx.pubsub.ItemsExtension;
+import org.jivesoftware.smackx.pubsub.LeafNode;
+import org.jivesoftware.smackx.pubsub.PayloadItem;
+import org.jivesoftware.smackx.pubsub.PubSubException.NotALeafNodeException;
+import org.jivesoftware.smackx.pubsub.PubSubManager;
+import org.jivesoftware.smackx.usertune.element.UserTuneElement;
+
+import org.jxmpp.jid.BareJid;
+import org.jxmpp.jid.EntityBareJid;
+
+/**
+ * Entry point for Smacks API for XEP-0118: User Tune.
+ * <br>
+ * To publish a UserTune, please use {@link #publishUserTune(UserTuneElement)} method. This will publish the node.
+ * <br>
+ * To stop publishing a UserTune, please use {@link #clearUserTune()} method. This will send a disabling publish signal.
+ * <br>
+ * To add a UserTune listener in order to remain updated with other users UserTune, use {@link #addUserTuneListener(UserTuneListener)} method.
+ * <br>
+ * To link a UserTuneElement with {@link Message}, use {@link #addUserTuneToMessage(Message, UserTuneElement)} method.
+ * <br>
+ * An example to illustrate is provided inside UserTuneElementTest inside the test package.
+ * <br>
+ * @see <a href="https://xmpp.org/extensions/xep-0118.html">
+ *     XEP-0118: User Tune</a>
+ */
+public final class UserTuneManager extends Manager {
+
+    public static final String USERTUNE_NODE = "http://jabber.org/protocol/tune";
+    public static final String USERTUNE_NOTIFY = USERTUNE_NODE + "+notify";
+
+    private static final Map<XMPPConnection, UserTuneManager> INSTANCES = new WeakHashMap<>();
+
+    private final Set<UserTuneListener> userTuneListeners = new HashSet<>();
+    private final AsyncButOrdered<BareJid> asyncButOrdered = new AsyncButOrdered<>();
+    private final PubSubManager pubSubManager;
+
+    public static synchronized UserTuneManager getInstanceFor(XMPPConnection connection) throws NotLoggedInException {
+        UserTuneManager manager = INSTANCES.get(connection);
+        if (manager == null) {
+            manager = new UserTuneManager(connection);
+            INSTANCES.put(connection, manager);
+        }
+        return manager;
+    }
+    private UserTuneManager(XMPPConnection connection) throws NotLoggedInException {
+        super(connection);
+        PepManager.getInstanceFor(connection).addPepListener(new PepListener() {
+            @Override
+            public void eventReceived(EntityBareJid from, EventElement event, Message message) {
+                if (!USERTUNE_NODE.equals(event.getEvent().getNode())) {
+                    return;
+                }
+
+                final BareJid contact = from.asBareJid();
+                asyncButOrdered.performAsyncButOrdered(contact, new Runnable() {
```

Comment:
Feel free to use a lambda here instead of a `Runnable`.

Reply:
I wanted to use lambdas since ages.
😌